projects/com.oracle.truffle.llvm.test/src/com/oracle/truffle/llvm/test/GCCBCTestSuite.java

```diff
@@ -0,0 +1,99 @@
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+ * GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.oracle.truffle.llvm.test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import com.oracle.truffle.llvm.LLVM;
+import com.oracle.truffle.llvm.runtime.LLVMLogger;
+import com.oracle.truffle.llvm.runtime.LLVMUnsupportedException;
+import com.oracle.truffle.llvm.runtime.LLVMUnsupportedException.UnsupportedReason;
+
+@RunWith(Parameterized.class)
+public class GCCBCTestSuite extends TestSuiteBase {
+
+    private static final int UNSIGNED_BYTE_MAX_VALUE = 0xff;
+    private TestCaseFiles tuple;
+    private File byteCodeFile;
+
+    public GCCBCTestSuite(TestCaseFiles tuple) {
+        this.tuple = tuple;
+        this.byteCodeFile = tuple.getBitCodeFile();
+    }
+
+    @Parameterized.Parameters
+    public static List<TestCaseFiles[]> getTestFiles() throws IOException {
+        File configFile = LLVMPaths.GCC_TEST_SUITE_CONFIG;
+        File testSuite = LLVMPaths.GCC_TEST_SUITE;
+        LLVMLogger.info("...start to read and compile files");
+        List<TestCaseFiles[]> files = getTestCasesFromConfigFile(configFile, testSuite, new TestCaseGeneratorImpl(false));
+        LLVMLogger.info("...finished reading and compiling files!");
+        return files;
+    }
+
+    @Test
+    public void test() {
+        try {
+            LLVMLogger.info("original file: " + tuple.getOriginalFile());
+            int expectedResult;
+            try {
+                expectedResult = TestHelper.executeLLVMBinary(byteCodeFile).getReturnValue();
+            } catch (Throwable t) {
+                t.printStackTrace();
+                throw new LLVMUnsupportedException(UnsupportedReason.CLANG_ERROR);
+            }
+            int truffleResult = truncate(LLVM.executeMain(byteCodeFile));
+            boolean undefinedReturnCode = tuple.hasFlag(TestCaseFlag.UNDEFINED_RETURN_CODE);
+            boolean pass = true;
+            if (!undefinedReturnCode) {
+                pass &= expectedResult == truffleResult;
+            }
+            recordTestCase(tuple, pass);
+            if (!undefinedReturnCode) {
+                Assert.assertEquals(byteCodeFile.getAbsolutePath(), expectedResult, truffleResult);
+            }
+        } catch (Throwable e) {
+            recordError(tuple, e);
+            throw e;
+        }
+    }
+
+    private static int truncate(int retValue) {
```

Comment:
Can we add this logic in `TestSuiteBase`? Otherwise, we have the same code in both GCC test suite classes.